tests/CustomLoggerTest.php

```diff
@@ -0,0 +1,231 @@
+<?php declare(strict_types=1);
+
+use PHPUnit\Framework\Attributes\{
+  Test,
+  CoversClass,
+  DataProvider,
+  UsesClass,
+};
+
+use PHPUnit\Framework\TestCase;
+use Monolog\Level;
+use Monolog\Handler\StreamHandler;
+use Bcgov\NaadConnector\CustomLogger;
+
+/**
+ * CustomLogger Class for testing CustomLogger class.
+ * This will test the class constructor, and _convertLogLevel
+ * which is a private method
+ *
+ * @category Client
+ * @package  NaadConnector
+ * @author   Richard O'Brien <Richard.OBrien@gov.bc.ca>
+ * @license  https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link     https://alerts.pelmorex.com/
+ */
+#[CoversClass('Bcgov\NaadConnector\CustomLogger')]
+#[UsesClass('Bcgov\NaadConnector\CustomLogger')]
+final class CustomLoggerTest extends TestCase
+{
+
+    /**
+     * Test the CustomLogger Constructor.
+     *
+     * @return void
+     */
+    #[Test]
+    public function testDefaultConstructor()
+    {
+        $logger = new CustomLogger();
+
+        // Check if the handlers are configured as expected.
+        $handlers = $logger->getHandlers();
+        $this->assertCount(1, $handlers);
+        $this->assertInstanceOf(StreamHandler::class, $handlers[0]);
+
+        // Assert the default logging level is 'info'
+        $streamHandler = $handlers[0];
+        $this->assertEquals(Level::Info, $streamHandler->getLevel());
+
+        // Assert that the default channel name is 'monolog'.
+        $this->assertEquals('monolog', $logger->getName());
+    }
+
+    /**
+     * Test the CustomLogger Constructor with parameters.
+     * this will instantiate the CustomLogger with:
+     * - channelName: 'my_channel'
+     * - logLevel: 'debug'
+     *
+     * @return void
+     */
+    #[Test]
+    public function testCustomConstructor()
+    {
+        $logger = new CustomLogger('my_channel', 'debug');
+
+        // Assert that the channel name has been set to 'my_channel'.
+        $this->assertEquals('my_channel', $logger->getName());
+
+        // Assert the default logging level is 'Debug'
+        $handlers = $logger->getHandlers();
+        $streamHandler = $handlers[0];
+        $this->assertEquals(Level::Debug, $streamHandler->getLevel());
+    }
+
+    /**
+     * Test the _convertLogLevel(string $loggingLevel) method
+     * to ensure it correctly converts the logging level to the Monolog Level.
+     *
+     * @return void
+     */
+    #[Test]
+    public function testLogLevelConversion()
+    {
+        $logLevels = $this->getLogLevelMappings();
+
+        foreach ($logLevels as $levelString => $expectedLevel) {
+            $logger = $this->createLoggerWithLevel($levelString);
```

Comment:
```suggestion
            $logger = new CustomLogger('test', $levelString);
```
Might be simpler, especially since now we have to write PHPDoc comments for every function here.